src/not_haskell.erl

```diff
@@ -0,0 +1,30 @@
+-module(not_haskell).
+
+-export([mapAccumL/3, for/2]).
+
+
+-spec mapAccumL(Fun, Acc, List) -> {Acc, List} when
+  %% boo args are backwards... why...
+  Fun :: fun((Elem :: T, Acc) -> {Acc, Out}),
+  Acc :: term(),
+  List :: [T],
+  Out :: term(),
+  T :: term().
+mapAccumL(Fun, Acc, List) ->
+  lists:foldl(fun(_x, {_acc, _outlist}) ->
```

Comment:
not the best to lead variable with _?
in erlang this supresses not used errors

Reply:
:o, that's a useful thing to know >:)

Reply:
haha, i saw this in your other code and though it was intentional

Reply:
http://erlang.org/pipermail/erlang-questions/2008-April/034468.html